cola/linalg/inverse/pseudo.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from plum import dispatch
+
+from cola.linalg.algorithm_base import Algorithm, Auto, IterativeOperatorWInfo
+from cola.linalg.inverse.gmres import LSTSQ_GMRES
+from cola.ops.operators import Diagonal, Identity, LinearOperator, Permutation, ScalarMul
+from cola.utils import export
+
+_SIZE = 1e6
+
+
+class LSTSQ(Algorithm):
+    """
+    Least-squares algorithm for computing the solve of a linear equation
+    """
+    def __call__(self, A: LinearOperator):
+        return LSTSQSolve(A)
+
+
+class LSTSQSolve(LinearOperator):
+    def __init__(self, A: LinearOperator):
+        super().__init__(A.dtype, A.shape)
+        self.A = A.to_dense()
+
+    def _matmat(self, X):
+        return self.xnp.lstsq(self.A, X)
+
+
+@export
+@dispatch.abstract
+def pseudo(A: LinearOperator, alg: Algorithm = Auto()):
```

Comment:
to make it more like numpy or scipy, pinv would be a more apt name I think

Reply:
Ok.